src/rupali/Shop.java

```diff
@@ -0,0 +1,105 @@
+package rupali;
+
+public class Shop {
+	
+	int stockofMaggie=50;
+	int stockofDosa=43;
+	int stockofOilPouches=39;
+	int stockofPanipuri=43;
+	int stockofmasala=73;
+	
+	void requestedQ(boolean reqType,int maggieq, int dosaQ, int oilQ, int panipuriQ, int masalaQ ){ 
+		if(maggieq==0) 
+			System.out.println("Sorry maggie running out of stock");
+		else 
+			  if(stockofMaggie>=maggieq) {
+				System.out.println("We served your request for maggie");
+				stockofMaggie=stockofMaggie-maggieq;
+			  }
+				else if(reqType==true) {	
+					System.out.println("We are providing partial request of maggie "+maggieq);
+					maggieq=0;
+				}	
+		
+		if(dosaQ==0) 
+			System.out.println("Sorry dosa running out of stock");
+		else if(stockofDosa>=dosaQ) {
+				System.out.println("We served your request for dosa packet");
+				stockofDosa=stockofDosa-dosaQ;
+		}
+				else if(reqType==true) {	
+					System.out.println("We are providing partial request of dosa "+dosaQ);
+					dosaQ=0;
+				}
+		if(oilQ==0) 
```

Comment:
Why we printing the OOS statements here, we have separate requirement for this FR2 and FR3